File: Raven.Client.Java/src/main/java/net/ravendb/client/utils/encryptors/IEncryptor.java
package net.ravendb.client.utils.encryptors;

import java.security.NoSuchAlgorithmException;
import java.security.spec.InvalidKeySpecException;


public interface IEncryptor {
  public IHashEncryptor createHash();

  public ISymmetricalEncryptor createSymmetrical(int keySize);

  public IAsymmetricalEncryptor createAsymmetrical(byte[] exponent, byte[] modulus) throws NoSuchAlgorithmException, InvalidKeySpecException;

}

File: src/Raven.Server/Smuggler/Migration/ApiKey/IEncryptor.cs
namespace Raven.Server.Smuggler.Migration.ApiKey
{
    public interface IEncryptor
    {
        IHashEncryptor Hash { get; }

        IHashEncryptor CreateHash();

        ISymmetricalEncryptor CreateSymmetrical();

        ISymmetricalEncryptor CreateSymmetrical(int keySize);

        IAsymmetricalEncryptor CreateAsymmetrical();

        IAsymmetricalEncryptor CreateAsymmetrical(byte[] exponent, byte[] modulus);

        IAsymmetricalEncryptor CreateAsymmetrical(int keySize);
    }
}